service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1256,6 +1262,8 @@ public void doRefresh() {
 
     @Override
     public void doCommit(TableMetadata base, TableMetadata metadata) {
+      eventEmitter.fire(new BeforeTableCommitEvent(base, metadata));
```

Comment:
One potential problem with the Jakarta events approach is that the firing is synchronous. This call can potentially take a long time to complete, depending on how many observers are subscribed, and how long they take to process the event.

There is a `fireAsync` method, I think it would be more appropriate.

But in general, maybe using a more robust event bus like the one provided by Vert.x would be a better idea.

Reply:
Vert.x event buses sound interesting, but I haven't used them before. Do you already have a sense of the tradeoffs or would you unequivocally suggest using them?